### ISSUE TITLE
Integrations are not garbage collected when deleting the syndesis CR

### ISSUE THREAD

**Issue by astefanutti** [syndesisio/syndesis#5033]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[*] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description

When deleting the syndesis CR, the deployed integrations are not deleted.

The integration deployment config resources should be owned by the syndesis platform, and garbage collected when the later is deleted.

**Comment by valdar** [comment]
probably also the camelk integration CRs needs to.

**Comment by astefanutti** [comment]
When using Camel K, maybe even the entire integration platform should be deleted and cascade delete the integration context and the integrations.

**Comment by valdar** [comment]
yep, I meant directly or indirectly through deleting whole camelk.
